permission_handler/android/src/main/java/com/baseflow/permissionhandler/ServiceManager.java

```diff
@@ -147,9 +150,16 @@ private static boolean isLocationServiceEnablePreKitKat(Context context)
                 Settings.Secure.LOCATION_PROVIDERS_ALLOWED);
         return !TextUtils.isEmpty(locationProviders);
     }
-
-    private boolean isBluetoothServiceEnabled() {
-        final BluetoothAdapter bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
-        return bluetoothAdapter.isEnabled();
+    // Suppress deprecation warnings since its purpose is to support to be backwards compatible with
+    // pre S versions of Android
+    @SuppressWarnings("deprecation")
+    private boolean isBluetoothServiceEnabled(Context context) {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            BluetoothManager manager = (BluetoothManager) context.getSystemService(BLUETOOTH_SERVICE);
+            final BluetoothAdapter adapter = manager.getAdapter();
+            return adapter.isEnabled();
+        } else {
+            return BluetoothAdapter.getDefaultAdapter().isEnabled();
+        }
```

Comment:
I feel we should reverse this `if` statement. So that the default behaviour would be to use the non-deprecated API and only use the deprecated API when the SDK version is below `JELLY_BEAN_MR2`:

```suggestion
        if (android.os.Build.VERSION.SDK_INT < android.os.Build.VERSION_CODES.JELLY_BEAN_MR2) {
            return BluetoothAdapter.getDefaultAdapter().isEnabled();
        } 
        
        BluetoothManager manager = (BluetoothManager) context.getSystemService(BLUETOOTH_SERVICE);
        final BluetoothAdapter adapter = manager.getAdapter();
        return adapter.isEnabled();
```